server/lib/src/idm/oauth2.rs

```diff
@@ -3680,6 +3688,70 @@ mod tests {
                 == Oauth2Error::InvalidOrigin
         );
 
+        // * invalid uri (doesn't match query params)
```

Comment:
Does the URL have to match query params? I would think its base url only

Reply:
If the query param may contain an open redirect, we can't allow it. We want to be secure.

Reply:
Are there any provisions in the rfc for this? I'd have to read through it again